aten/src/ATen/native/mps/operations/LossOps.mm

```diff
@@ -371,7 +362,9 @@ static void nllnd_loss_backward_impl(Tensor& grad_input_arg,
                                                             name:@"scaleByWeightTensor"];
       }
       if (reduction == Reduction::Mean) {
-        oneHotTensor = divisionNoNaN(mpsGraph, oneHotTensor, totalWeightTensor);
+        oneHotTensor = [mpsGraph divisionWithPrimaryTensor:oneHotTensor
```

Comment:
Hmm, this change feels wrong to me, can you explain why do you need it for the fix?

Reply:
I wasn't sure about this.

I can keep it as it was. One concern I have is that if I can remove code without any tests breaking, the code isn't tested. We should add tests for this. Can you think of a case where we'd get nans here and I can add it?

Reply:
Perhaps there even already exist tests elsewhere and we can enable them as in #134184.

Reply:
Did some digging. Looks like the only way we can get here with `total_weight = 0` is with an empty input tensor? It's otherwise guarded upstream from what I can tell. I might be wrong though.

Looking at the implementation for CUDA and CPU I don't see the same safe division. Doesn't hurt to leave it there, but it should be a noop IIUC.

Reply:
Resolving, but please let me know if you have any additional insights.